src/main/java/org/aksw/simba/lemming/creation/Inferer.java

```diff
@@ -189,6 +253,96 @@ public OntModel readOntology(String filePath) {
 
 		return ontModel;
 	}
-	
-	
+
+	/**
+	 * Searches for the equivalents in an ontology and maps them to our Equivalent<T
+	 * extends OntResource> class, producing a map of the Equivalent objects with
+	 * the URIs as keys.
+	 * 
+	 * @see Equivalent<T extends OntResource>
+	 * @param <T>
+	 * @param ontElements the ontology classes or properties
+	 * @return
+	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	public <T extends OntResource> Map<String, Equivalent> searchEquivalents(Set ontElements) {
+
+		Map<String, Equivalent> uriNodeMap = new HashMap<String, Equivalent>();
+
+		Stack<T> stack = new Stack<T>();
+		stack.addAll(ontElements);
+
+		Map<String, Boolean> visitedMap = (Map<String, Boolean>) ontElements.stream()
+				.collect(Collectors.toMap(Object::toString, visited -> false));

```

Comment:
I don't think that this map is really necessary. Wouldn't it sufficient to check `!uriNodeMap.contains(current.toString())`?

Reply:
yes, you are right! I've changed it :)